src/java/org/apache/cassandra/utils/btree/BTree.java

```diff
@@ -3174,23 +3163,23 @@ void copy(Object[] unode, int usz, int offset, int length)
          * {@code from} may be {@code -1}, representing the first child only;
          * all other indices represent the key/child pairs that follow (i.e. a key and its right-hand child).
          */
-        private void copyNoOverflow(Object[] unode, int usz, int[] uszmap, int offset, int length)
+        private void copyPrecedingNoOverflow(Object[] unode, int usz, int[] uszmap, int offset, int length)
         {
             if (length <= 1)
             {
                 if (length == 0)
                     return;
 
+                buffer[MAX_KEYS + count] = unode[usz + offset];
+                sizes[count] = uszmap[offset] - (offset > 0 ? (1 + uszmap[offset - 1]) : 0);
                 buffer[count] = unode[offset];
-                buffer[MAX_KEYS + count + 1] = unode[usz + offset + 1];
-                sizes[count + 1] = uszmap[offset + 1] - (1 + uszmap[offset]);
                 ++count;
             }
             else
             {
+                System.arraycopy(unode, usz + offset, buffer, MAX_KEYS + count, length);
```

Comment:
Not super important, just wonder why you flipped the order here, so that we access the latter half of the array first?